### DISCUSSION TITLE
Is sqlite-viewer-app closed source? / WAL mode

### DISCUSSION THREAD

**Discussion by dourgulf** [qwtel/sqlite-viewer-vscode#54] [Ideas]
I want to do some improve about Wal Mode.

**Comment by qwtel** [comment]
The parts that run in vscode with elevated permissions are open. The sandboxed/UI parts are closed.

I can grant access in certain cases, such as if you have a concrete proposal. Wrt to WAL mode, I don't think there are any easy wins. The extension is using a WASM build of sqlite and vscode itself is running in Node.js, which doesn't have any shared memory primitives. It would need a custom VFS at minimum, but without shared memory it basically can't be done properly. 

For readonly mode to at least show the latest transactions from the WAL file is something I'm looking into right now, but I suspect it will involve diving into sqlite's c source. If you have insights there, feel free to reach out.
